spec/authoring_spec.ts

```diff
@@ -716,6 +724,62 @@ describe('authoring', () => {
+        el(['content-create']).click();
+        el(['content-create-dropdown']).element(by.buttonText('editor3 template')).click();
+        browser.wait(ECE.visibilityOf(el(['authoring-field--media-gallery', 'media-gallery--upload-placeholder'])));
+        expect(ECE.hasElementCount(els(['authoring-field--media-gallery', 'media-gallery-image']), 0)()).toBe(true);
+
+        el(['media-gallery--upload-placeholder']).click();
+
+        browser.wait(ECE.presenceOf(el(['image-upload-input'])));
+        selectFilesForUpload(el(['image-upload-input']), ['image-big.jpg']);
+
+        el(['media-metadata-editor', 'field--headline'], by.tagName('input')).sendKeys('image headline');
+        el(['media-metadata-editor', 'field--alt_text'], by.tagName('input')).sendKeys('image alt text');
+        el(['media-metadata-editor', 'field--description_text'], by.tagName('textarea')).sendKeys('image description');
+
+        el(['multi-image-edit--start-upload']).click();
+
+        waitAndClick(el(['change-image', 'done']));
+
+        assertToastMsg('success', 'Item updated.');
+        waitForToastMsgDissapear('success', 'Item updated.');
+
+        browser.wait(ECE.hasElementCount(els(['authoring-field--media-gallery', 'media-gallery-image']), 1), 1000);
+
+        hover(el(['authoring-field--media-gallery', 'media-gallery-image']));
+        el(['media-gallery-image--edit']).click();
+        el(['done']).click(); // click done without making any changes
+
+        assertToastMsgNotDisplayed('success', 'Item updated.');
+        expect(authoring.save_button.isEnabled()).toBe(false);
+    });
+
+    it('Can add an image with default crops to media gallery', () => {
+        workspace.selectDesk('XEditor3 Desk'); // has media gallery in content profile
+        el(['content-create']).click();
+        el(['content-create-dropdown']).element(by.buttonText('editor3 template')).click();
+        browser.wait(ECE.visibilityOf(el(['authoring-field--media-gallery', 'media-gallery--upload-placeholder'])));
+        expect(ECE.hasElementCount(els(['authoring-field--media-gallery', 'media-gallery-image']), 0)()).toBe(true);
+
+        el(['media-gallery--upload-placeholder']).click();
+
+        browser.wait(ECE.presenceOf(el(['image-upload-input'])));
+        selectFilesForUpload(el(['image-upload-input']), ['image-big.jpg']);
+
+        el(['media-metadata-editor', 'field--headline'], by.tagName('input')).sendKeys('image headline');
+        el(['media-metadata-editor', 'field--alt_text'], by.tagName('input')).sendKeys('image alt text');
+        el(['media-metadata-editor', 'field--description_text'], by.tagName('textarea')).sendKeys('image description');
+
+        el(['multi-image-edit--start-upload']).click();
+
+        waitAndClick(el(['change-image', 'done']));
+
+        browser.wait(ECE.hasElementCount(els(['authoring-field--media-gallery', 'media-gallery-image']), 1), 1000);
```

Comment:
Is there a reason for a specific timeout? I always go with the default, because when the element is found, the test will not wait any longer anyway. And if the test fails, it doesn't matter much if it fails after 5 or 10 seconds. Or does it? If you have many tests failing, it would take longer for tests to run, but I'd say a reasonable default solves this.

Reply:
I don't remember why I put that there, probably just something I forgot to remove